src/components/appExit/DockerContainerShutdown.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect, useState } from 'react';
+import { ipcRenderer } from 'electron';
+import { useStoreActions } from 'store';
+import { useAsyncCallback } from 'react-async-hook';
+import { Modal, Spin } from 'antd';
+import { LoadingOutlined } from '@ant-design/icons';
+import { usePrefixedTranslation } from 'hooks';
+
+const DockerContainerShutdown: React.FC = () => {
+  const [isShuttingDown, setIsShuttingDown] = useState(false);
+  const { stopAll } = useStoreActions(s => s.network);
+  const stopDockerContainersAsync = useAsyncCallback(stopAll);
+  const { l } = usePrefixedTranslation('cmps.appExit.DockerContainerShutdown');
+  useEffect(() => {
+    ipcRenderer.on('app-closing', async () => {
+      setIsShuttingDown(true);
+      await stopDockerContainersAsync.execute();
+    });
+    return () => {
+      ipcRenderer.removeAllListeners('app-closing');
+    };
+  }, []);
+
+  return (
+    <>
+      <Modal open={isShuttingDown} closable={false} maskClosable={false} footer={null}>
+        <div
+          style={{
+            display: 'flex',
+            justifyContent: 'center',
+            alignItems: 'center',
+            height: '100%',
+          }}
+        >
```

Comment:
We use the `emotion` CSS-in-JS library for styling in this project. Can you move these styles in a `Styled` mapping? Take a look at other components for examples of how it should look.